### PR TITLE
Changes to ease AMP Compliance

### PR DESCRIPTION
**Pull request by mpereira96**

### CODE REVIEW

File: src/aol-hb.js [hunk 1]
```diff
@@ -2,7 +2,8 @@ import 'helpers/polyfills';
 import BidManager from 'bidManager';
 import RenderingManager from 'renderingManager';
 
-let globalContext = window.aolhb = {};
+let globalContext = window.aolhb = window.aolhb || {};
+globalContext.que = globalContext.que || [];
```
Comment (slimkrazy) [inline src/aol-hb.js]: que -> queue

**Comment by slimkrazy** [comment]
Minor comment on the spelling of 'queue'. LGTM outside of that.

File: src/aol-hb.js [hunk 2]
```diff
@@ -27,4 +32,27 @@ globalContext.init = (bidRequestConfig, placementsConfigs) => {
   };
 };
 
+globalContext.que.push = (cmd) => {
```
Comment (marian-r) [inline src/aol-hb.js]: Are you overriding array method here? Because I don't think this implementation will add the command to the array.

File: src/aol-hb.js [hunk 3]
```diff
@@ -27,4 +32,27 @@ globalContext.init = (bidRequestConfig, placementsConfigs) => {
...
+  if (typeof cmd === 'function') {
+    try {
+      cmd.call();
+    } catch (e) {
+      console.warn('Error processing command :' + e.message);
```
Comment (marian-r) [inline src/aol-hb.js]: It would be a good idea to dump the whole error object to the console, the same as you did below, i.e. `console.warn('Error processing command', e);`.

File: src/aol-hb.js [hunk 4]
```diff
@@ -27,4 +32,27 @@ globalContext.init = (bidRequestConfig, placementsConfigs) => {
...
+  }
+};
+
+for (let i = 0; i < globalContext.que.length; i++) {
+  if (typeof globalContext.que[i].called === 'undefined') {
+    try {
+      globalContext.que[i].call();
+      globalContext.que[i].called = true;
+    } catch (e) {
+      console.warn('Error processing command :', 'aol-hb.js', e);
```
Comment (marian-r) [inline src/aol-hb.js]: I would suggest removing space before the colon. Having `aol-hb.js` in the console message is a nice idea. Do we want to apply that to all console logs?

File: src/aol-hb.js [hunk 5]
```diff
@@ -27,4 +32,27 @@ globalContext.init = (bidRequestConfig, placementsConfigs) => {
   };
 };
 
+globalContext.queue.push = (cmd) => {
+  if (typeof cmd === 'function') {
+    try {
+      cmd.call();
+    } catch (e) {
+      console.warn('Error processing command', 'aol-hb.js', e);
```
Comment (marian-r) [inline src/aol-hb.js]: I had the same issue with understanding this, see my previous comment :D But the idea is that this method is overridden after the array is already populated.

Comment (mpereira96) [inline src/aol-hb.js]: 1. Theres a possibility that this queue has been initiated as an empty array in the client side in the same global variable page `window.aolhb` before the library has been loaded, so I could initiate as an empty array or as an existent array (that will have functions to execute) from the client side.

2. Yes I override the push method to execute functions, so if client side still uses this queue after the library has been loaded, it straight away executes any function that is pushed.

3. Yes after the library has been loaded and on initiating, I will see if theres any functions to execute on the queue that the user previously added on the client side.

The goal is to allow the user to queue functions before the library has been loaded. In an AMP scenario, in the `remote.html` page, the library is being loaded asynchronously as every resource should be according to AMP standards. 

So on `draw3p`, where magic needs to happen, the library hasn't been loaded yet so we're providing the publishers a easy way to queue code. 
This is also good functionality to have in a normal case, because it allows publishers load the library asynchronously and they can carry on loading the rest of the page.

File: src/aol-hb.js [hunk 6]
```diff
@@ -27,4 +32,27 @@ globalContext.init = (bidRequestConfig, placementsConfigs) => {
   };
 };
 
+globalContext.queue.push = (cmd) => {
+  if (typeof cmd === 'function') {
+    try {
+      cmd.call();
```
Comment (mpereira96) [inline src/aol-hb.js]: I think it makes more sense to call it `.call` because you are calling (executing the function) at this point in time. `.called` seems more like it would be a flag or a property.

Comment (marian-r) [inline src/aol-hb.js]: I think @vzhukovsky was asking why you are not setting the `called` property just like [below](https://github.com/aol/aol-hb.js/pull/16/files#diff-e8024450d0d2f5327ea857a09562e878R51).

Comment (mpereira96) [inline src/aol-hb.js]: Oh okay my bad, I'm not pushing it to the array, this is called straight away from the `.push` function. The bellow scenario is to prevent the weird case if the library being initiated more than one time.

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "aol-hb",
-  "version": "0.9.4",
```
Comment (vzhukovsky) [inline package.json]: Why did you change it?

Comment (mpereira96) [inline package.json]: Because it is a change in the library and we have to publish as a newer version correct?

Comment (mpereira96) [inline package.json]: Actually I should have bumped the minor version instead.
